File: fluentcore/src/main/java/com/mcreater/fxfluent/util/listeners/NewValueListener.java
package com.mcreater.fxfluent.util.listeners;

import javafx.beans.value.ChangeListener;
import javafx.beans.value.ObservableValue;

public interface NewValueListener<T> extends ChangeListener<T> {
    default void changed(ObservableValue<? extends T> observableValue, T t, T t1) {
        changed(t1);
    }
    void changed(T t1);
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/util/listeners/NewValueListener.kt
package com.mcreater.fxfluent.util.listeners

import javafx.beans.value.ChangeListener
import javafx.beans.value.ObservableValue


fun interface NewValueListener<T> : ChangeListener<T> {
    override fun changed(observableValue: ObservableValue<out T>, t: T, t1: T) {
        changed(t1)
    }

    fun changed(t1: T)
}

